File: ios/MyNaitveModule.m
//
//  MyNaitveModule.m
//  swiftturbomodules
//
//  Created by TransformHub on 01/07/22.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <UIKit/UIKit.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(MyNaitveModule, NSObject)

RCT_EXTERN_METHOD(myNaitveMethod:(NSString*)text  resolve: (RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/MyNaitveModule.swift
//
//  MyNaitveModule.swift
//  swiftturbomodules
//
//  Created by TransformHub on 01/07/22.
//

import Foundation
import React

@objc(MyNaitveModule)
class MyNaitveModule: NSObject {
  var myNaitveResolve: RCTPromiseResolveBlock!
  var myNaitveReject: RCTPromiseRejectBlock!
  
  @objc static func requiresMainQueueSetup() -> Bool {
    return false
  }
  
  @objc func myNaitveMethod(_ text: String,resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
    myNaitveResolve = resolve
    myNaitveReject = reject
    let resultsDict = [
      "msg" : "Hello \(text)"
    ];
    self.myNaitveResolve(resultsDict)
  }
}














































